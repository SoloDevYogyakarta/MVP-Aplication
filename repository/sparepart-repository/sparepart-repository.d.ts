import { SparepartInstance } from '../../database/entities/services/sparepart-entity/sparepart-entity';
export declare class SparepartRepository {
    private readonly logger;
    findALl(): Promise<SparepartInstance[]>;
    findOne(id: number): Promise<SparepartInstance>;
}
