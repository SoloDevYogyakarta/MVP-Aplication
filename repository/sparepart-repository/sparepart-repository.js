"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var SparepartRepository_1;
Object.defineProperty(exports, "__esModule", { value: true });
exports.SparepartRepository = void 0;
const common_1 = require("@nestjs/common");
const sparepart_associate_1 = require("../../database/associates/sparepart-associate/sparepart-associate");
const sparepart_entity_1 = require("../../database/entities/services/sparepart-entity/sparepart-entity");
let SparepartRepository = SparepartRepository_1 = class SparepartRepository {
    constructor() {
        this.logger = new common_1.Logger(SparepartRepository_1.name);
    }
    async findALl() {
        this.logger.log(SparepartRepository_1.name);
        return await sparepart_associate_1.sparepartAssociate.findAll({
            attributes: sparepart_associate_1.sparepartAttribute,
            include: sparepart_associate_1.sparepartInclude,
        });
    }
    async findOne(id) {
        this.logger.log(SparepartRepository_1.name);
        return await sparepart_entity_1.sparepartEntity.findOne({
            where: { id },
            attributes: sparepart_associate_1.sparepartAttribute,
            include: sparepart_associate_1.sparepartInclude,
        });
    }
};
exports.SparepartRepository = SparepartRepository;
exports.SparepartRepository = SparepartRepository = SparepartRepository_1 = __decorate([
    (0, common_1.Injectable)()
], SparepartRepository);
//# sourceMappingURL=sparepart-repository.js.map