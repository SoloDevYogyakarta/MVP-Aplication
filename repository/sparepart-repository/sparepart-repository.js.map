{"version":3,"file":"sparepart-repository.js","sourceRoot":"","sources":["../../../src/repository/sparepart-repository/sparepart-repository.ts"],"names":[],"mappings":";;;;;;;;;;AAAA,2CAAoD;AACpD,sDAAmD;AACnD,2GAI2E;AAC3E,yGAG4E;AAGrE,IAAM,mBAAmB,2BAAzB,MAAM,mBAAmB;IAAzB;QACY,WAAM,GAAG,IAAI,eAAM,CAAC,qBAAmB,CAAC,IAAI,CAAC,CAAC;IAsBjE,CAAC;IApBC,KAAK,CAAC,OAAO,CAAC,KAAa,EAAE,IAAY;QACvC,IAAI,KAAK,GAAG,EAAE,CAAC;QACf,IAAI,CAAC,MAAM,CAAC,GAAG,CAAC,qBAAmB,CAAC,IAAI,CAAC,CAAC;QAC1C,KAAK,GAAG,IAAA,eAAM,EAAC,KAAK,EAAE,IAAI,CAAC,CAAC;QAC5B,OAAO,MAAM,wCAAkB,CAAC,OAAO,CAAC;YACtC,KAAK,EAAE,KAAK;YACZ,UAAU,EAAE,wCAAkB;YAC9B,OAAO,EAAE,sCAAgB;YACzB,KAAK,EAAE,CAAC,CAAC,IAAI,EAAE,KAAK,CAAC,CAAC;SACvB,CAAC,CAAC;IACL,CAAC;IAED,KAAK,CAAC,OAAO,CAAC,EAAU;QACtB,IAAI,CAAC,MAAM,CAAC,GAAG,CAAC,qBAAmB,CAAC,IAAI,CAAC,CAAC;QAC1C,OAAO,MAAM,kCAAe,CAAC,OAAO,CAAC;YACnC,KAAK,EAAE,EAAE,EAAE,EAAE;YACb,UAAU,EAAE,wCAAkB;YAC9B,OAAO,EAAE,sCAAgB;SAC1B,CAAC,CAAC;IACL,CAAC;CACF,CAAA;AAvBY,kDAAmB;8BAAnB,mBAAmB;IAD/B,IAAA,mBAAU,GAAE;GACA,mBAAmB,CAuB/B"}