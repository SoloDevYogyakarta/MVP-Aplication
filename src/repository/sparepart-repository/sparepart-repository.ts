import { Injectable, Logger } from '@nestjs/common';
import {
  sparepartEntity,
  SparepartInstance,
} from '../../database/entities/services/sparepart-entity/sparepart-entity';

@Injectable()
export class SparepartRepository {
  private readonly logger = new Logger(SparepartRepository.name);

  async findALl(): Promise<SparepartInstance[]> {
    this.logger.log(SparepartRepository.name);
    return await sparepartEntity.findAll();
  }

  async findOne(id: number): Promise<SparepartInstance> {
    this.logger.log(SparepartRepository.name);
    return await sparepartEntity.findOne({ where: { id } });
  }
}
