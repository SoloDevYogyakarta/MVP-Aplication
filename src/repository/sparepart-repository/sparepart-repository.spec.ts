import { SparepartRepository } from './sparepart-repository';

describe('SparepartRepository', () => {
  let repository: SparepartRepository;

  beforeEach(() => {
    repository = new SparepartRepository();
  });

  it('should to be defined', () => expect(repository).toBeDefined());

  it('render correctly', () => expect(repository).toMatchSnapshot());
});
