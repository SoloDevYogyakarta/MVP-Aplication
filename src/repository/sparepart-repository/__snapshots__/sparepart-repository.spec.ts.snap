// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SparepartRepository render correctly 1`] = `
SparepartRepository {
  "logger": Logger {
    "context": "SparepartRepository",
    "options": {},
  },
}
`;
